source/tutorials/kubernetes.rst

```diff
@@ -3,354 +3,724 @@
       sudo systemctl enable kubelet.service
 
-#. Disable swap using one of the following methods, either:
 
-   a) Temporarily:
+.. important::
+
+   If network proxy settings are required for Internet connectivity, configure
+   them now because the scripts will propagate proxy configuration based on
+   the running configuration. It is especially important to set the
+   :command:`no_proxy` variable for Kubernetes. See the `Setting proxy servers
+   for Kubernetes`_ section for details.
+
 
-      .. code-block:: bash
+Initialize the master node
+**************************
+
+In Kubernetes, a master node is part of the `Kubernetes Control Plane
+<https://kubernetes.io/docs/concepts/#kubernetes-control-plane>`_. 
 
-         sudo swapoff -a
+Initializing a new Kubernetes cluster involves crafting a :command:`kubeadm
+init` command. Adding parameters to this command can control the fundamental
+operating components of the cluster. This means it is important to understand
+and choose network and runtime options before running a :command:`kubeadm
+init` command.
 
-      .. note::
 
-         Swap will be enabled at next reboot, causing failures in
-         your cluster if you choose this method.
+Choose a pod network add-on
+===========================
 
-   or:
+See `What is a Container Network Interface (CNI)?`_ for information on what
+pod network add-ons and CNIs.
 
-   b) Permanently:
+It is important to decide which CNI will be used early because some pod
+network add-ons require configuration during cluster initialization. Check
+whether or not your add-on requires special flags when you initialize the
+master control plane.
 
-      Mask the swap partition:
+If your chosen network add-on requires appending to the :command:`kubeadm
+init` command, make note of it before continuing. For example, if you choose
+the *flannel* pod network add-on, then in later steps you must add the
+following to the :command:`kubeadm init` command:
 
-      .. code-block:: bash
+.. code-block:: console
+
+   --pod-network-cidr 10.244.0.0/16
 
-         sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null
-         sudo swapoff -a
+.. important::
 
-      .. note::
+   The version of CNI plugins installed needs to be compatible with the
+   version of Kubernetes that is installed otherwise the cluster may fail.
+   Check the Kubernetes version with :command:`kubeadm version -o short` and
+   refer to the documentation of the CNI plugins to obtain a compatible
+   version.
 
-         On systems with limited resources, some performance degradation may
-         be observed while swap is disabled.
+Choose a container runtime
+==========================
 
-#. Switch to root to modify the `hosts` file:
+See `What is a Container Runtime Interface (CRI)?`_ for more information on
+what a CRI is.
+
+|CL| supports Kubernetes with the various runtimes
+below with or without `Kata Containers`_:
+
+* `CRI+O`_
+* `containerd`_
```

Comment:
I've seen a lot of sites capitalize containerd when it stands alone and have seen other sites not capitalize. I don't know which way is correct.

Reply:
I'll leave this up to TCS in a follow up editorial PR.

Reply:
It's fine as is. CNCF defines this: https://containerd.io/